alfred-telemetry-platform/src/main/java/eu/xenit/alfred/telemetry/binder/solr/sharding/SolrShardingMetricsContainer.java

```diff
@@ -45,18 +45,20 @@ public void refresh() {
     }
 
     public Set<ShardInstance> getShardInstances(Shard shard) {
-        return rawData.get(shard.getFloc()).get(shard).stream().map(shardState -> shardState.getShardInstance())
+        return rawData.get(shard.getFloc()).get(shard).stream().map(shardState -> (shardState != null ? shardState.getShardInstance():null)).filter(instance -> instance != null)
                 .collect(
                         Collectors.toSet());
     }
 
     public ShardState getShardState(ShardInstance shardInstance) {
         Shard shard = shardInstance.getShard();
         return rawData.get(shard.getFloc()).get(shard).stream()
-                .filter(shardState -> shardState.getShardInstance() == shardInstance).findAny().get();
+                .filter(shardState -> shardState.getShardInstance() == shardInstance).findAny().orElse(null);
     }
 
     public ReplicaState getReplicaState(ShardInstance shardInstance) {
+        ShardState shardState = getShardState(shardInstance);
+        if(shardState == null) return null;
```

Comment:
And the `Optional<ShardState>` propagates here to return `Optional<ReplicateState>`